spartan_parallel/src/scalar/mod.rs

```diff
@@ -42,10 +42,15 @@ pub trait SpartanExtensionField:
   + Hash
   + From<Self::InnerType>
   + fmt::Debug
+  + Mul<Self::BaseField, Output = Self>
+  + MulAssign<Self::BaseField>
 {
   /// Inner Goldilocks extension field
   type InnerType: ExtensionField + Field;
 
+  /// Basefield for conserving computational resources
+  type BaseField: SpartanExtensionField;
```

Comment:
`BaseField` itself is not `SpartanExtensionField`. why not use `type BaseField: Field`?

Reply:
You're right. This is corrected.